mmrotate/datasets/pipelines/transforms.py

```diff
@@ -97,25 +97,46 @@ class PolyRandomRotate(object):
     Reference: https://github.com/hukaixuan19970627/OrientedRepPoints_DOTA
 
     Args:
-        rate (bool): (float, optional): The rotating probability.
+        rotate_ratio (float, optional): The rotating probability.
             Default: 0.5.
-        angles_range(int, optional): The rotate angle defined by random
-            (-angles_range, +angles_range).
+        mode (str, optional) : Indicates whether the angle is chosen in a
+            random range (mode='range') or in a preset list of angles
+            (mode='value'). Defaults to 'range'.
+        angles_range(int|list[int], optional): The range of angles.
+            If mode='range', angle_ranges is an int and the angle is chosen
+            in (-angles_range, +angles_ranges).
+            If mode='value', angles_range is a non-empty list of int and the
+            angle is chosen in angles_range.
+            Defaults to 180 as default mode is 'range'.
         auto_bound(bool, optional): whether to find the new width and height
             bounds.
         rect_classes (None|list, optional): Specifies classes that needs to
             be rotated by a multiple of 90 degrees.
-        version  (str, optional): Angle representations. Defaults to 'oc'.
+        version  (str, optional): Angle representations. Defaults to 'le90'.
     """
 
     def __init__(self,
                  rotate_ratio=0.5,
+                 mode='range',
                  angles_range=180,
                  auto_bound=False,
                  rect_classes=None,
                  version='le90'):
         self.rotate_ratio = rotate_ratio
         self.auto_bound = auto_bound
+        if mode == 'range':
+            if not isinstance(angles_range, int):
+                raise TypeError("PolyRandomRotate with mode 'range' \
+                                  expects angle_range to be an int.")
+        elif mode == 'value':
+            if not (isinstance(angles_range, list)
+                    and isinstance(angles_range[0], int)):
+                raise TypeError("PolyRandomRotate with mode 'value' expects \
+                                  angle_range to be a non-empty list of int.")
+        else:
+            raise ValueError(f"Unknown mode for PolyRandomRange: {mode}. \
+                               Mode is supposed to be 'range' or 'value'.")
```

Comment:
We can direct limit the mode by `assert`.
assert mode in ['range', 'vaule], f"`model` is supposed to be 'range' or 'value', but get {model}."